src/lib.rs

```diff
@@ -35,3 +35,34 @@ pub use crate::config::{
 pub use crate::subscriber::{
     configure_subscriber, init_subscriber, ContextInfo, EventFormatter, Tracing, Uninstall,
 };
+
+/// Create a tracing error event, casting the error to &dyn [std::error::Error] for [layer::ErrorLayer],
+/// and adding the type name as error.kind.
+///
+/// Usage:
+/// ```
+/// use prima_tracing::report_error;
+///
+/// let error = "not a number".parse::<usize>().unwrap_err();
+/// report_error!(&error, "Parsing error!");
+/// ```
+///
+/// You can also add use add attributes and do things, just like with a regular [tracing::error]
+/// macro call
+/// ```
+/// use prima_tracing::report_error;
+/// # let input = "not a number";
+/// # let uid = "1223";
+///
+/// let error = input.parse::<usize>().unwrap_err();
+/// report_error!(&error, input, user=uid, "Parsing error: {}", &error);
+/// ```
+#[macro_export]
+macro_rules! report_error {
+    ($error:expr, $($args:tt)*) => {
+        {
+          let kind = ::std::any::type_name_of_val(&$error);
+          ::tracing::error!(error.kind = kind, error = $error as &dyn ::std::error::Error, $($args)+)
```

Comment:
I think (not tested) that it should be like this:
```suggestion
          $crate::tracing::error!(error.kind = kind, error = $error as &dyn ::std::error::Error, $($args)+)
```
With the current code I'm getting the following error when a crate doesn't depend on `tracing` directly:
```
error[E0433]: failed to resolve: could not find `tracing` in the list of imported crates
  --> src/config/src/lib.rs:11:5
   |
11 |     report_error!(error, "Parsing error");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `tracing` in the list of imported crates
   |
   = note: this error originates in the macro `report_error` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
in that case we also need to reexport tracing

Reply:
Right! Which I think is a good idea anyway, especially given how important it is to match all of the otel crates versions

Reply:
Done